context.go

```diff
@@ -141,7 +141,7 @@ func (context *Context) DialConfig(config *websocket.Config) (*Sender, error) {
 			}
 		}
 	}()
-	context.activeSubProtocol = selectedSubProtocol
+	context.activeSubProtocol = ExtractAppProtocolId(selectedSubProtocol)
```

Comment:
This isn't your change, but makes sense to do it as part of this PR - I don't think `ExtractAppProtocolId` should be an exported function.

Reply:
Sure, that sounds reasonable.